docs/develop-guideline.md

```diff
@@ -0,0 +1,120 @@
+- Pulling: it pulls the index, manifest, config and layers of a image, which is dependent on the crate [image-rs](https://github.com/confidential-containers/image-rs)
+- Unpacking: it decrypts and decompresses the layers of a image if necessary. The decryption is dependent on
+the crate [ocicrypt-rs](https://github.com/confidential-containers/ocicrypt-rs).
+- Storing: it saves the unpacking layers on an encrypted filesystem
+- Mounting: it merges layers to a rootfs for a application container and mounts the rootfs to a mounting point
+
+## Security Management
+The enclave agent need to satisfy security requirements, including
+- attestation
+    - remote attestation: it provides security information for the remove 
+    - local attestation
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native attesation client
+- signature verification: 
+    - some images are signed by the pulisher or the store or both, it need to verify these signatures
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native signature client 
+
+# 2. Development Guide
+
+
+
+## Occlum-based Enclave Agent
+
+### *Dependencies*
+
+#### **rats-tls**
+The [rats-tls](https://github.com/inclavare-containers/inclavare-containers/tree/master/rats-tls) is a library for communicating on a secured channel, which supports the native attestation agent with feature `eaa kbc`.
+
+The rats-tls depends on the SGX SDK (following the [installing guide](develop-env-install.md)), libraries `libsgx-dcap-ql` and `libsgx-dcap-ql-dev` (install them if necessary).
+
+The rats-tls provides a different version for the application on the occlum or on the host. We want to build and run a KBC (Key Broken Service) supporting `eaa kbc` on the host, we should build and install rats-tls for the host. The library will be installed in `/usr/local/lib/rats-tls`.
+```bash
+git clone https://github.com/inclavare-containers/rats-tls.git
+cd rats-tls
+cmake -DRATS_TLS_BUILD_MODE="host" -DBUILD_SAMPLES=on -H. -Bbuild
+make -C build install
+```
+
+If we want to build the library for application running on the occlum, we need to build an occlum version. 
+```bash
+git clone https://github.com/inclavare-containers/rats-tls.git
+cd rats-tls
+cmake -DRATS_TLS_BUILD_MODE="occlum" -DBUILD_SAMPLES=on -H. -Bbuild
+make -C build install
+```
+The library is also installed in the `/usr/local/lib/rats-tls` in default. If we want to keep two versions on the same machine, we need to install the occlum version instance into a customized place `[rats-tls-path]` and refer it in a config of the occlum instance.
+
+### *Agent Bundle Building*
+There are three steps to build a occlum-based enclave agent. Copy following three scripts under directory `enclave-cc/src` and run the command to create the bundle.
```

Comment:
Please unify the markdown format.
For example: the title and the file are separated by one line